core/src/main/java/com/gentics/mesh/core/data/impl/UserImpl.java

```diff
@@ -223,6 +224,12 @@ public User setEmailAddress(String emailAddress) {
 		return new TraversalResult<>(out(ASSIGNED_TO_ROLE).frameExplicit(RoleImpl.class));
 	}
 
+	@Override
+	public Page<? extends Role> getRolesViaShortcut(User user, PagingParameters params) {
+		VertexTraversal<?, ?, ?> traversal = out(ASSIGNED_TO_ROLE);
```

Comment:
Check whether it is possible to use the index instead. This would speedup the GraphQL field retrieval for `.roles`.

Reply:
```
// GraphQL:
String indexName = "e." + ASSIGNED_TO_ROLE + "_out";
// userId: this.id()
new DynamicTransformablePageImpl<>(ac.getUser(), indexName, userId, RoleImpl.class, pagingInfo, perm, null, true);
```

Reply:
Done